src/der.rs

```diff
@@ -197,6 +213,57 @@ mod tests {
         assert!(r.is_err());
     }
 
+    type FileLines<'a> = std::io::Lines<std::io::BufReader<&'a std::fs::File>>;
+
+    fn read_pem_section(lines: & mut FileLines, section_name: &str)
+                        -> std::vec::Vec<u8> {
+        // Skip comments and header
+        let begin_section = format!("-----BEGIN {}-----", section_name);
+        loop {
+            let line = lines.next().unwrap().unwrap();
+            if line == begin_section {
+                break;
+            }
+        }
+
+        let mut base64 = std::string::String::new();
+
+        let end_section = format!("-----END {}-----", section_name);
+        loop {
+            let line = lines.next().unwrap().unwrap();
+            if line == end_section {
+                break;
+            }
+            base64.push_str(&line);
+        }
+
+        base64.from_base64().unwrap()
+    }
+
```

Comment:
NIT: Please put take the added code that is currently above this comment, and move it to be below test_parse_bad_spki_der. This way, the code is sorted top-down.